src/pages/[...slug].tsx

```diff
@@ -96,7 +97,6 @@ export const getStaticProps: GetStaticProps<Props, Params> = async (
 
   const markdown = getContentBySlug(`${locale}/${params.slug.join("/")}`)
   const frontmatter = markdown.frontmatter
-  const tocItems = markdown.tocItems
```

Comment:
Basically instead of running this inside `getContentBySlug`, we'll wait for that and a few other things to finish which gives us a compiled string which includes the exact heading elements that have been found.

We then feed that to an updated `generateTableOfContents` that parses based on this more accurate array of heading data.